Samples/Swift/nhubsample/nhubsample/NotificationDetailViewController.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  Copyright © 2018 Microsoft All rights reserved.
+//  Licensed under the Apache License (2.0).
+//
+
+import UIKit
+
+class NotificationDetailViewController: UIViewController {
+    @IBOutlet weak var titleLabel: UILabel!
+    @IBOutlet weak var dismissButton: UIButton!
+    @IBOutlet weak var bodyLabel: UILabel!
+    
+    var _userInfo: [String: Any]?
+    
+    init(_ userInfo: Any) {
+        super.init(nibName: "NotificationDetail", bundle: nil)
+        _userInfo = userInfo as? [String: Any]
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder);
+    }
+    
+    override func viewDidLayoutSubviews() {
+        self.titleLabel.sizeToFit();
+        self.bodyLabel.sizeToFit();
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        var title: String?
+        var body: String?
+    
+        let aps = _userInfo?["aps"] as? [String: Any]
+        let alertObject = aps?["alert"]
+        if (alertObject != nil) {
+            if let alertDict = alertObject as? [String: Any] {
+                title = alertDict["title"] as? String
```

Comment:
The strings should be moved to the constants enums. No magic strings

Reply:
We are interacting with a JSON dictionary. I'd rather keep the strings inline, so that it's easier to read. This is just a sample and users can decide if they want magic strings or not.

If there is a better way of interacting with a dynamic object, I'd be interested in improving this code in general. I'm not a fan of using dictionaries for this.